src/config.ml

```diff
@@ -1297,6 +1318,38 @@ let parse_next (effect:parser_effect) initial_state : (parser_state, 'err) resul
           in
           (* TODO consult `Proto_force *)
           retb (B(Remote, [host, port, proto]))
+        | (`Dev _ | `Dev_type _) as current ->
+          (* there must be a corresponding `Dev or `Dev_type in [tl]*)
+          let typs, tl = List.partition (function
+              | `Dev_type _ -> true
+              | _ -> false) (current::tl) in
+          let names, tl = List.partition (function
+              | `Dev _ -> true
+              | _ -> false) tl in
```

Comment:
the two `List.partition` increase the time complexity of `loop` (instead of O(n), it is now O(3 * n)). fine with me, an alternative could be to pass dev/dev-type as argument to loop (similar to acc).

Reply:
done as you suggested; it's also much easier to read :)